modules/ensemble/lib/widget/webview/native/webviewstate.dart

```diff
@@ -183,40 +54,247 @@ class WebViewState extends EWidgetState<EnsembleWebView> with CookieMethods {
     super.didUpdateWidget(oldWidget);
   }
 
+  String _generateHeaderOverrideScript() {
+    return '''
+      const headerRules = ${jsonEncode(widget.controller.headerOverrideRules.map((r) => r.toJson()).toList())};
+
+      function matchesRule(url, rule) {
+        switch(rule.matchType) {
+          case 'HeaderMatchType.CONTAINS':
+            return url.includes(rule.urlPattern);
+          case 'HeaderMatchType.EXACT':
+            return url === rule.urlPattern;
+          case 'HeaderMatchType.REGEX':
+            return new RegExp(rule.urlPattern).test(url);
+          default:
+            return false;
+        }
+      }
+
+      const originalFetch = window.fetch;
+      window.fetch = async function(url, options = {}) {
+        options = options || {};
+        options.headers = options.headers || {};
+        
+        for (const rule of headerRules) {
+          if (matchesRule(url, rule)) {
+            
+            options.headers = rule.mergeExisting 
+              ? {...options.headers, ...rule.headers}
+              : {...rule.headers};
+              
+          }
+        }
+          
+        return originalFetch(url, options);
+      };
+    ''';
+  }
+
+  Future<void> setCookie(io.Cookie? cookie) async {
+    if (widget.controller.url == null) {
+      return;
+    }
+    final mainUrl = Uri.parse(widget.controller.url!);
+    if (cookie != null) {
+      final cookieDomain = cookie.domain ?? mainUrl.host;
+
+      // Set cookie for main domain
+      await widget.controller.cookieManager.setCookie(
+        url: Uri.parse('https://$cookieDomain'),
```

Comment:
we can't assume that it's always https. Please get the protocol from the url as well and don't hardcode it